src/view/com/util/List.web.tsx

```diff
@@ -83,13 +86,60 @@ function ListImpl<ItemT>(
     })
   }
 
+  const getScrollableNode = React.useCallback(() => {
+    if (containWeb) {
+      if (!nativeRef.current) return null
+      return {
+        scrollWidth: nativeRef.current?.scrollWidth,
+        scrollHeight: nativeRef.current?.scrollHeight,
+        clientWidth: nativeRef.current?.clientWidth,
+        clientHeight: nativeRef.current?.clientHeight,
+        scrollY: nativeRef.current?.scrollTop,
+        scrollX: nativeRef.current?.scrollLeft,
+        scrollTo(options?: ScrollToOptions) {
+          nativeRef.current?.scrollTo(options)
+        },
+        scrollBy(options: ScrollToOptions) {
+          nativeRef.current?.scrollBy(options)
+        },
+        addEventListener(event: string, handler: any) {
+          nativeRef.current?.addEventListener(event, handler)
+        },
+        removeEventListener(event: string, handler: any) {
+          nativeRef.current?.removeEventListener(event, handler)
+        },
+      }
+    } else {
+      return {
+        scrollWidth: document.documentElement.scrollWidth,
+        scrollHeight: document.documentElement.scrollHeight,
+        clientWidth: window.innerWidth,
+        clientHeight: window.innerHeight,
+        scrollY: window.scrollY,
+        scrollX: window.scrollX,
+        scrollTo(options: ScrollToOptions) {
+          window.scrollTo(options)
+        },
+        scrollBy(options: ScrollToOptions) {
+          window.scrollBy(options)
+        },
+        addEventListener(event: string, handler: any) {
+          window.addEventListener(event, handler)
+        },
+        removeEventListener(event: string, handler: any) {
+          window.removeEventListener(event, handler)
+        },
+      }
+    }
+  }, [containWeb])
```

Comment:
This is the main thing to pay attention to. It feels safer to create a a single source of truth on what a function does or a value returns rather than have a bunch of `if (containWeb) {} else {}` all over the component.